packages/remix-dev/compiler/plugins/serverEntryModulePlugin.ts

```diff
@@ -45,6 +45,7 @@ ${Object.keys(config.routes)
   export const assetsBuildDirectory = ${JSON.stringify(
     config.relativeAssetsBuildDirectory
   )};
+  export const features = ${JSON.stringify(config.features)};
```

Comment:
How are these going to be surfaced to the browser at runtime? Are they going to be serialized into the EntryContext? This is part of why I suggested the "read at build time and inject into globals" approach.

Reply:
Ah yeah ok. Seems like I still prefer build time for this for a few reasons:
1) no increase in client bundle size as we can optimize out unused code paths
2) no increase in serialized context size. this also exposes internals about your site configuration to the public, something I'm not a fan of

Reply:
@jacob-ebey We discussed these tradeoffs, but @ryanflorence is afraid of overloading `process.env` when we're pushing towards supporting a zero-build environment. That's a larger conversation, but I deferred to the product manager for now 😅

Reply:
@brophdawg11 was on the call too. Sometimes a feature flag might only change one or two lines of behavior so no need to duplicate all of that in the function, but most importantly just want to keep everything "the same". It's a runtime thing in all places we need it: bundling, server, client.